### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by gurpreetatwal**
Update some of the packages the sdk relies on, started out updating xmlbuilder only added the rest in the interest of consolidating the updates into one PR
### Updated:
- xmlbuilder 2.6.2 -> 4.2.1
- sax 1.1.5 -> 1.2.1
- xml2js 0.4.15 -> 0.4.17
- uuid 3.0.0 -> 3.0.1
### xmlbuilder

The main reason for updating xmlbuilder is that the current version relies on lodash 3, which although not deprecated is an older release. Most projects that depend on lodash will most likely install 4 and installing aws-sdk-js in their project will install lodash 3 alongside lodash 4.

I also didn't update to the latest version because version 5 and up [drop support for `node < 0.10`](https://github.com/oozcitak/xmlbuilder-js/commit/586433a144777926341cafb2b4c691a1d1f1621a) and aws-sdk supports `node >= 0.8`. Allthough upgrading to the latest version wouldn't be the worst thing, it would just spit out a warning which `escodegen` currently does as well.

According to the [xmlbuilder changelog](https://github.com/oozcitak/xmlbuilder-js/blob/master/CHANGELOG.md) there don't seem to be any changes that would break the sdk. I also ran the unit tests locally and they all passed. 
### sax changes
- fix bug when object is used as a chunk
- add new event
- [changes between 1.1.5 and 1.2.1](https://github.com/isaacs/sax-js/compare/v1.1.5...master)
### xml2js changes
- added option to allow whitespace only text-nodes
- update xmlbuilder dep to 8.2.0  
  - Note: this will cause the warning to show for `node < 0.10`
- [changes between 0.4.15 and master](https://github.com/Leonidas-from-XIV/node-xml2js/compare/0.4.15...master)

### uuid
- no meaningful changes, internal code reorg
- [changes](https://github.com/kelektiv/node-uuid/compare/v3.0.0...v3.0.1)

### CODE REVIEW

**Comment by chrisradek** [comment]
@gurpreetatwal 
Thanks for the PR! Really appreciate that you provided links to meaningful changes between dependency versions as well. We'll take a look.

**Comment by gurpreetatwal** [comment]
@chrisradek  any updates?

**Comment by edrpls** [comment]
@gurpreetatwal This fork is now giving me the error:

```
Error: Cannot find module 'lodash/object/assign'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/edited/node_modules/xmlbuilder/lib/index.js:5:12)
    at Object.<anonymous> (/edited/node_modules/xmlbuilder/lib/index.js:14:4)
    at Module._compile (module.js:570:32)
    at Module._extensions..js (module.js:579:10)
    at Object.require.extensions.(anonymous function) [as .js] (/edited/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/edited/node_modules/aws-sdk/lib/xml/builder.js:2:15)
    at Module._compile (module.js:570:32)
```

I'm going to try and debug this but wanted to give you a heads up.

**Comment by edrpls** [comment]
Looks like the commit from two days ago rolled back all the changes to package.json

**Comment by chrisradek** [comment]
Sorry, it turns out merging these dependencies has not been straightforward because of some compatibility issues with older versions of node that we still support. I haven't had time lately to dive deeper into resolving them yet, but I should have some more time this coming week.

File: package.json [hunk 1]
```diff
@@ -1,77 +1,119 @@
...
+  "dependencies": {
+    "buffer": "4.9.1",
+    "crypto-browserify": "1.0.9",
+    "jmespath": "0.15.0",
+    "querystring": "0.2.0",
+    "sax": "1.1.5",
+    "url": "0.10.3",
+    "uuid": "3.0.0",
+    "xml2js": "0.4.15",
+    "xmlbuilder": "2.6.2"
```
Comment (edrpls) [inline package.json]: The versions are the same as master now.

Comment (gurpreetatwal) [inline package.json]: whoops, my bad

When I merged in the changes from master, the entire `package.json` file got replaced and I neglected to update the versions

**Comment by goldenbearkin** [comment]
what is stopping this merge?

**Comment by shyamchandranmec** [comment]
@chrisradek Any update on when this will be merged?

**Comment by prabhatsharma** [comment]
While we wait for Chris and team to fix it, I have temporarily solved this by installing latest lodash and xmlbuilder manually in my package.json. Am using aws-sdk on front end and not on node.

**Comment by baweaver** [comment]
Primarily it comes down to bundle size and having multiple versions of Lodash hanging around. 

Thankfully the require technologies being used circumvent that issue quite nicely, otherwise most of Node could be crashed with 3 versions of Lodash and a few of Underscore :)

**Comment by prabhatsharma** [comment]
There is no concern on _ variable in global scope. Everything seems to be playing well. Indeed with tree shaking functionality on front end code bundle size is also not really a problem. Its just an unnecessary minor annoyance for now. On the server side code, package size is generally not a  concern. So for now all looks good.

**Pull request closed without merging**